wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -511,6 +493,47 @@ public TypeSpec generateMessage(MessageType type) {
     return builder.build();
   }
 
+  /**
+   * Preallocate all of the names we'll need for this type. Names are allocated in precedence
+   * order, so names we're stuck with (serialVersionUID etc.) occur before proto field names are
+   * assigned. Names we aren't stuck with (typically for locals) yield to message fields.
+   */
+  private NameAllocator allocateNames(MessageType type) {
+    NameAllocator nameAllocator = new NameAllocator();
+    nameAllocator.newName("serialVersionUID", "serialVersionUID");
+    nameAllocator.newName("ADAPTER", "ADAPTER");
+    nameAllocator.newName("MESSAGE_OPTIONS", "MESSAGE_OPTIONS");
+    if (emitAndroid) {
+      nameAllocator.newName("CREATOR", "CREATOR");
+    }
+    Set<String> collidingNames = collidingFieldNames(type.fieldsAndOneOfFields());
+    for (Field field : type.fieldsAndOneOfFields()) {
+      String suggestion = collidingNames.contains(field.name())
+          ? field.qualifiedName()
+          : field.name();
+      nameAllocator.newName(suggestion, field);
```

Comment:
quite happy that name allocator automatically replaces the dots in the qualified name with underscores